### PR TITLE
enable `noImplicitAny` and disable `skipLibCheck` in TS compiler (closes #2497,#2457)

### LINKED ISSUE #2497: error TS7006: Parameter 'req' implicitly has an 'any' type
### Are you requesting a feature or reporting a bug?
Bug 

### What is the current behavior?

In TypeScript project, tsc will produce the following errors (due to the ` "noImplicitAny": true` in the tsc configuration file):

```sh
node_modules/testcafe/ts-defs/index.d.ts(921,54): error TS7006: Parameter 'req' implicitly has an 'any' type.
node_modules/testcafe/ts-defs/index.d.ts(921,59): error TS7006: Parameter 'res' implicitly has an 'any' type.
node_modules/testcafe/ts-defs/index.d.ts(928,40): error TS7006: Parameter 'req' implicitly has an 'any' type.
node_modules/testcafe/ts-defs/index.d.ts(928,45): error TS7006: Parameter 'res' implicitly has an 'any' type.
node_modules/testcafe/ts-defs/index.d.ts(1573,73): error TS7006: Parameter 'req' implicitly has an 'any' type.
node_modules/testcafe/ts-defs/index.d.ts(1573,78): error TS7006: Parameter 'res' implicitly has an 'any' type.
```

tsc runs well after doing the following modifications:
```js
interface RequestMock {
    /**
     * Specifies requests to intercept
     * @param filter - Specifies which requests should be mocked with a response that follows in the `respond` method.
     */
    onRequestTo(filter: string | RegExp | object | ((req: any, res: any) => boolean)): RequestMock;
    /**
     * Specifies the mocked response.
     * @param body - The mocked response body.
     * @param statusCode - The response status code.
     * @param headers - Custom headers added to the response in the property-value form.
     */
    respond(body?: object | string | ((req: any, res: any) => any), statusCode?: number, headers?: object): RequestMock;
}

export function RequestLogger(filter?: string | RegExp | object | ((req: any, res: any) => boolean), options?: RequestLoggerOptions): RequestLogger;

```

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system: Windows 10
* testcafe version: 0.20.3
* node.js version: v10.1.0
* typescript: 2.6.1
* tsconfig.json
```json
    /* Strict Type-Checking Options */
    "strict": true,                   
    "noImplicitAny": true,            
    "strictNullChecks": true,         
    "strictFunctionTypes": true,      
    "noImplicitThis": true,           
    "alwaysStrict": true,             
```

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

File: src/compiler/test-file/formats/typescript/compiler.js [hunk 1]
```diff
@@ -17,14 +17,14 @@ export default class TypeScriptTestFileCompiler extends APIBasedTestFileCompiler
             allowJs:                 true,
             pretty:                  true,
             inlineSourceMap:         true,
-            noImplicitAny:           false,
+            noImplicitAny:           true,
```
Comment (AndreyBelym) [inline src/compiler/test-file/formats/typescript/compiler.js]: We shouldn't enable strict type checking options until we provide to a user a way to disable them.

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**